### PR TITLE
[SYCL] Adapt to sycl 2020 exceptions

### PR DESCRIPTION
**Pull request by pwisniewskimobica**
Due to issue https://github.com/intel/llvm/issues/7268
I changed old exceptions to new ones (`sycl::exception`)

### CODE REVIEW

File: sycl/source/backend.cpp [hunk 1]
```diff
@@ -38,8 +38,7 @@ static const PluginPtr &getPlugin(backend Backend) {
   case backend::ext_oneapi_cuda:
     return pi::getPlugin<backend::ext_oneapi_cuda>();
   default:
-    throw sycl::runtime_error{"getPlugin: Unsupported backend",
-                              PI_ERROR_INVALID_OPERATION};
+    throw sycl::exception(sycl::errc::runtime, "getPlugin: Unsupported backend");
```
Comment (steffenlarsen) [inline sycl/source/backend.cpp:41]: I wonder what constructor of `sycl::exception` it uses for the current version? Either way, I would feel better if we explicitly convert it to a `std::error_code` using `sycl::make_error_code`:

```suggestion
    throw sycl::exception(sycl::make_error_code(sycl::errc::runtime), "getPlugin: Unsupported backend");
```

**Review by steffenlarsen — COMMENTED** [review]
Thanks a lot for making these changes! I just have a small request and then it needs to be formatted with `clang-format`, then I'm happy with it. 😄

**Review by steffenlarsen — APPROVED** [review]
Great, thank you! There seems to be some failing tests due to them expecting the old exceptions, but otherwise it LGTM!

**Comment by pwisniewskimobica** [comment]
@steffenlarsen thanks
Do you think I also need to fix those tests? Or leave at it is at let you or someone to merge this PR?

**Comment by steffenlarsen** [comment]
> @steffenlarsen thanks Do you think I also need to fix those tests? Or leave at it is at let you or someone to merge this PR?

We need the tests to pass before merging the PR, otherwise it will make other PRs and the post-commit testing fail these tests too.

**Comment by pwisniewskimobica** [comment]
@steffenlarsen Thanks for comment
I fixed tests related to my change and also included them in separate commit in PR.
Unfortunately as you wrote there  are some old e2e tests failing which are not trivial for me to fix as they are not related.
Please take a look 
thanks

**Comment by steffenlarsen** [comment]
> @steffenlarsen Thanks for comment I fixed tests related to my change and also included them in separate commit in PR. Unfortunately as you wrote there are some old e2e tests failing which are not trivial for me to fix as they are not related. Please take a look thanks

Thank you, @pwisniewskimobica ! At first glance I don't think any of these failures are related to this patch. Normally it may help to push a merge with tip onto the this patch and let CI run again, though we have been seeing some odd failures in CI today so I cannot promise that it will fix it fully.

**Comment by pwisniewskimobica** [comment]
@steffenlarsen thanks
What you propose for me about this PR?

**Comment by steffenlarsen** [comment]
With the amount of failures in the current version I would prefer if you push a merge commit and hopefully it will at least reduce the number of failures we see. My worry is that if we consider the current failures as unrelated we may miss some that are in fact related.

**Comment by steffenlarsen** [comment]
So it did! Thanks again for making these changes.

**Pull request merged**